custom_components/climate_group/climate.py

```diff
@@ -180,33 +181,78 @@ def should_poll(self) -> bool:
     
     async def async_update(self):
         """Query all members and determine the climate group state."""
-        all_states = [self.hass.states.get(x) for x in self._entity_ids]
-        states = list(filter(None, all_states))
-        self._target_temp = _reduce_attribute(states, 'temperature')
-        self._current_temp = _reduce_attribute(states, 'current_temperature')
-        self._min_temp = _reduce_attribute(states, 'min_temp', reduce=max)
-        self._max_temp = _reduce_attribute(states, 'max_temp', reduce=min)
-        self._mode = None
-        all_modes = [state.state for state in states]
-        if all_modes:
-            # Report the most common effect.
-            modes_count = Counter(itertools.chain(all_modes))
-            self._mode = modes_count.most_common(1)[0][0]
+        raw_states = [self.hass.states.get(x) for x in self._entity_ids]
+        all_states = list(filter(None, raw_states))
+
+        states = list(filter(lambda x: x is not None and 'preset_mode' in x.attributes, raw_states))
+
+        heat_states = list(filter(lambda x: x.state == HVAC_MODE_HEAT, states))
+        cool_states = list(filter(lambda x: x.state == HVAC_MODE_HEAT, states))
+
+        heat_actions = list(filter(lambda x: x.attributes['hvac_action'] == CURRENT_HVAC_HEAT, states))
+        cool_actions = list(filter(lambda x: x.attributes['hvac_action'] == CURRENT_HVAC_COOL, states))
+        idle_actions = list(filter(lambda x: x.attributes['hvac_action'] == CURRENT_HVAC_IDLE, states))
+
+        preset_away_stats = list(filter(lambda x: x.attributes['preset_mode'] == PRESET_AWAY, states))
+        preset_none_stats = list(filter(lambda x: x.attributes['preset_mode'] != PRESET_AWAY, states))
+        self._is_away = len(preset_away_stats) == len(states)
+
+        if self._is_away:
+            self._target_temp = _reduce_attribute(preset_away_stats, 'temperature')
+            self._current_temp = _reduce_attribute(preset_away_stats, 'current_temperature')
+        else:
+            self._target_temp = _reduce_attribute(preset_none_stats, 'temperature')
+            self._current_temp = _reduce_attribute(preset_none_stats, 'current_temperature')
+
+        self._min_temp = _reduce_attribute(all_states, 'min_temp', reduce=max)
+        self._max_temp = _reduce_attribute(all_states, 'max_temp', reduce=min)
+
+        self._mode = CURRENT_HVAC_OFF
+        if len(heat_states):
+            self._mode = HVAC_MODE_HEAT
+        elif len(cool_states):
+            self._mode = HVAC_MODE_COOL
+
+        self._action = CURRENT_HVAC_OFF
+        if len(heat_actions):
```

Comment:
So if any thermostat is heating we will report `heat`,
again why not report the most common one?

Reply:
if you're using this for a multi-room grouping, and one thermostat is on, in my mind it is more logical to show that the heating is on _somewhere_, rather than idle. If I'm heating the living room only, the general heating is still "on", even if not on everywhere. Even in your original use case of 2 radiators in 1 room, isn't it more logical to show the group as "on" if anything is "on"? This is how the other HA groups work, e.g. switches. "On" if any element is on, "off" if they are all off.

I tried to put a logical order of precedence on them, for the relevance of the states.

Reply:
Indeed, I agree that HEAT and COOL should have precedence over idle.